src/build/dlopen.c

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file dlopen.c
+ * @brief Proof of concept for calling dlopen() from GNU make.
+ *
+ * @author James Jones (jejones@networkradius.com)
+ * @copyright 2020 Network RADIUS SARL (legal@networkradius.com)
+ */
+
+#include <string.h>
+#include <dlfcn.h>
+#include <gnumake.h>
+
+#ifdef __APPLE__
+#  define DL_EXTENSION ".dylib"
+#elif defined (WIN32)
+#  define DL_EXTENSION ".dll"
+#else
+#  define DL_EXTENSION ".so"
+#endif
+
+char	*mk_dlopen(char const *nm, unsigned int argc, char **argv);
+int	mk_dlopen_gmk_setup(void);
+
+int plugin_is_GPL_compatible;
+
+/** GNU make-callable layer around dlopen().
+ * 
+ * @return NULL on failure; otherwise, a string. 
+ * 
+ * @note argv[0] should point to the name of the file to open, minus the extension.
+ * 
+ */
+char * mk_dlopen(UNUSED char const *nm, UNUSED unsigned int argc, char **argv)
+{
+	void	*handle;
+	char	*extended_file_name;
+	char	*result;
+	
+	extended_file_name = gmk_alloc(1 + strlen(DL_EXTENSION) + strlen(argv[0]));
+	sprintf(extended_file_name, "%s%s", argv[0], DL_EXTENSION);
+
+	handle = dlopen(extended_file_name, RTLD_LAZY);
+
+	if (!handle) return NULL;
+	
+	result = gmk_alloc(2);
+	strcpy(result, "1");
+	return result;
+}
+
+/** Register function(s) with make.
+ * 
+ * @return non-zero value on success, or zero on failure.
+ * @note gmk_add_function() "returns" void, so we can't really say whether it
+ *       succeeded or failed. Thus the return of the constant 1.
+ */
+int mk_dlopen_gmk_setup(void)
+{
+	gmk_add_function("mk_dlopen", &mk_dlopen, 1, 1, 0);
```

Comment:
There's no reason to have the exported name `mk_dlopen`.  It can just be `dlopen`